ssl.c

```diff
@@ -256,13 +256,25 @@ redisSSLContext *redisCreateSSLContextWithOptions(redisSSLOptions *options, redi
     if (ctx == NULL)
         goto error;
 
-    ctx->ssl_ctx = SSL_CTX_new(SSLv23_client_method());
+    const SSL_METHOD *ssl_method;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
```

Comment:
Perhaps move the check var (0x10100000L) into a const - to keep it in one place?